python/cog/server/worker.py

```diff
@@ -153,10 +153,13 @@ def __init__(
         self._events = events
         self._tee_output = tee_output
         self._cancelable = False
+        self._events_lock = None
 
         super().__init__()
 
     def run(self) -> None:
+        self._events_lock = multiprocessing.Lock()
+
```

Comment:
I'm seeing a new [CI failure for typechecking](https://github.com/replicate/cog/actions/runs/9638832220/job/26665632298?pr=1758#step:5:22)

> Object of type "None" cannot be used with "with" (reportOptionalContextManager)

I think that could be fixed by doing assignment in `__init__`.

```suggestion
        self._events_lock = multiprocessing.Lock()

        super().__init__()

    def run(self) -> None:
```

Reply:
Or else, you could annotate as `Optional` in the initializer. Downside of that approach is that you'd have to add conditionals to check for non-`None` wherever that's used.

```suggestion
        self._events_lock: Optional[multiprocessing.Lock] = None

        super().__init__()

    def run(self) -> None:
        self._events_lock = multiprocessing.Lock()

```

Reply:
Thanks @mattt . I didn't notice that there was a type checker.
I have moved the lock initialization to the `__init__` and specified the lock for spawn, because  `_ChildWorker` is a spawn process. The default `multiprocessing.Lock()` is based on OS and can be `fork` lock, so it may not work if we move the lock initialization to `__init__` for some OS.